uaclient/log.py

```diff
@@ -58,3 +62,35 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+def get_user_log_file() -> str:
+    """Gets the correct user log_file storage location
+
+    returns XDG_CACHE_HOME env variable, if it is set,
+    If not set, we return the default path
+    """
+    user, _ = system.subp(["whoami"])
+    user = user.strip()
```

Comment:
I think we can use `os.getlogin()`

Reply:
This is after testing if I change users in the terminal using `su -l <user>` and noticed `os.getlogin()` returned the my user name instead of the other user I logged in as. i.e. if my user name is `user1` and I change the user in the terminal to `user2` 
`os.getlogin()` returns `user1`.

Reply:
Out of band conversation we decided to use getpass.getuser()